manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -136,6 +137,51 @@ public Processor createProcessor(String bridgeId, String customerId, ProcessorRe
         return newProcessor;
     }
 
+    @Override
+    @Transactional
+    public Processor updateProcessor(String processorId, String bridgeId, String customerId, ProcessorRequest processorRequest) {
+        // Attempt to load the Bridge. We cannot update a Processor if the Bridge is not available.
+        bridgesService.getReadyBridge(bridgeId, customerId);
+
+        // Extract existing definition
+        Processor existingProcessor = getProcessor(processorId, bridgeId, customerId);
+        ProcessorDefinition existingDefinition = jsonNodeToDefinition(existingProcessor.getDefinition());
+        String existingTransformationTemplate = existingDefinition.getTransformationTemplate();
+        BaseAction existingAction = existingDefinition.getRequestedAction();
+        BaseAction existingResolvedAction = existingDefinition.getResolvedAction();
+
+        // Validate update. Only Filters are supported at present.
+        if (!Objects.equals(existingProcessor.getName(), processorRequest.getName())) {
+            throw new BadRequestException("It is not possible to update the Processor's name.");
+        }
+        if (!Objects.equals(existingTransformationTemplate, processorRequest.getTransformationTemplate())) {
+            throw new BadRequestException("It is not possible to update the Processor's TransformTemplate.");
+        }
+        if (!Objects.equals(existingAction, processorRequest.getAction())) {
+            throw new BadRequestException("It is not possible to update the Processor's Action.");
+        }
+
+        // Create new definition copying existing properties
+        existingProcessor.setSubmittedAt(ZonedDateTime.now());
+        existingProcessor.setStatus(ManagedResourceStatus.ACCEPTED);
+
+        Set<BaseFilter> updatedFilters = processorRequest.getFilters();
+        ProcessorDefinition updatedDefinition = new ProcessorDefinition(updatedFilters, existingTransformationTemplate, existingAction, existingResolvedAction);
+        existingProcessor.setDefinition(definitionToJsonNode(updatedDefinition));
+
+        // Processor and Work should always be created in the same transaction
+        // Since updates to the Action are unsupported we do not need to update the Connector record.
+        processorDAO.persist(existingProcessor);
```

Comment:
This isn't required as you've already loaded the `Processor` into the `PersistenceContext` with the `getProcessor` method call.

Reply:
Removed... as you can tell my JPA is a little rusty.